### PR TITLE
Integration of HAL_TMR

### PR DESCRIPTION
**Pull request by mariuscrsn**
#5 @pastorpflores  Removed HRTIM from HAL_TMR_Clock_e. It not belongs to HAL_TMR

### CODE REVIEW

File: firmware/Sources/HAL/HAL_TMR/hal_tmr.h [hunk 1]
```diff
@@ -0,0 +1,153 @@
+/*********************************************************************************
+* @file           : hal_tmr.h
+* @brief          : HAL header file for timers
+**********************************************************************************
+* @attention
```
Comment (pastorpflores) [inline firmware/Sources/HAL/HAL_TMR/hal_tmr.h:5]: Headeeeeeers

File: firmware/Sources/Test/HAL/hal_tmr_test.c [hunk 2]
```diff
@@ -0,0 +1,146 @@
+/*********************************************************************************
+* @file           : hal_tmr.c
+* @brief          : Implementation of HAL TMR
+**********************************************************************************
+* @attention
```
Comment (pastorpflores) [inline firmware/Sources/Test/HAL/hal_tmr_test.c:5]: headeeeeers

**Review by pastorpflores — CHANGES_REQUESTED** [review]
Please change your templates

File: firmware/Sources/HAL/HAL_TMR/hal_tmr.c [hunk 3]
```diff
@@ -0,0 +1,150 @@
+/*********************************************************************************
+* @file           : hal_tmr.c
+* @brief          : Implementation of HAL TMR
+**********************************************************************************
+* @attention
```
[the remark this answers is not in the document]
Comment (mariuscrsn) [reply firmware/Sources/HAL/HAL_TMR/hal_tmr.c]: Done

File: firmware/Sources/Test/HAL/hal_tmr_test.c (hunk 2 above)
Comment (mariuscrsn) [reply firmware/Sources/Test/HAL/hal_tmr_test.c]: Deleted

File: firmware/Sources/HAL/HAL_TMR/hal_tmr.h (hunk 1 above)
Comment (mariuscrsn) [reply firmware/Sources/HAL/HAL_TMR/hal_tmr.h]: Doneeeeeeee

**Review by pastorpflores — APPROVED** [review]

**Pull request merged**